govcd/cse_test.go

```diff
@@ -222,22 +261,170 @@ func (vcd *TestVCD) Test_Cse(check *C) {
 			clusterSettings.WorkerPools[0].Name: {
 				MachineCount: 1,
 			},
-			"new-pool": {
+			"new-pool-1": {
 				MachineCount: 0,
 			},
+			"new-pool-2": {
+				MachineCount: 0, // Should remove autoscaler
+			},
 		},
 	}, true)
 	check.Assert(err, IsNil)
 	check.Assert(cluster.ControlPlane.MachineCount, Equals, 1)
 	for _, pool := range cluster.WorkerPools {
-		if pool.Name == "new-pool" {
+		if pool.Name == "new-pool-1" {
+			check.Assert(pool.MachineCount, Equals, 0)
+			check.Assert(pool.Autoscaler, IsNil)
+		} else if pool.Name == "new-pool-2" {
 			check.Assert(pool.MachineCount, Equals, 0)
+			check.Assert(pool.Autoscaler, IsNil)
 		} else {
 			check.Assert(pool.MachineCount, Equals, 1)
 		}
 	}
 }
 
+// Test_CseWithAutoscaler tests the autoscaling capabilities in CSE clusters
+func (vcd *TestVCD) Test_CseWithAutoscaler(check *C) {
```

Comment:
As this feature is not trivial, suggesting to extend/add a test to cover mis-configurations, etc. (the "non-happy" path).

Reply:
Added `Test_CseValidationErrors`. Actually I found+fixed a tiny validation bug thanks to this new test 🙂